src/main/java/nextstep/subway/domain/FavoriteRepository.java

```diff
@@ -0,0 +1,14 @@
+package nextstep.subway.domain;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface FavoriteRepository {
+    <S extends Favorite> S save(S entity);
```

Comment:
JpaRepository에 대한 의존이 사라지고
infra 패키지에 JpaFavoriteRepository가 추가되었군요 💯 

save()는 아래와 같이 수정 가능할것 같습니다 😃
```suggestion
public interface FavoriteRepository {
    Favorite save(Favorite favorite);
```